### PR TITLE
CondFD Conductivity/Specific Heat Actuator Bug

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
 - Fixes a bug in the CondFD conductivity and specific heat actuators where the k and cp values were not being updated in all places.
 - Minor doc updates.

### CODE REVIEW

File: doc/ems-application-guide/src/ems-actuators/thermal-envelope.tex [hunk 1]
```diff
@@ -110,7 +110,7 @@ \subsection{Conduction Finite Difference}\label{conduction-finite-difference}
 Several actuators, called ``CondFD Surface Material Layer,'' are available with actuated component control types for models that use the Conduction Finite Difference solution for conduction through surfaces. These are described below.
 
 \begin{itemize}
-    \item ``Thermal Conductivity''. Has units of [W/m-K]. Controls the thermal conductivity of the specified material layer within a given surface. Actuator is named from the surface name and material layer name combined with a colon ":". E.g. "SURFNAME:MATERIALLAYERNAME"
+    \item ``Thermal Conductivity''. Has units of [W/m-K]. Controls the thermal conductivity of the specified material layer within a given surface. Actuator is named from the surface name and material layer name combined with a colon ``:''. E.g. ``SURFNAME:MATERIALLAYERNAME''
```
Comment (mitchute) [inline doc/ems-application-guide/src/ems-actuators/thermal-envelope.tex:113]: Updated to use correct quotation marks.

File: doc/input-output-reference/src/overview/group-surface-construction-elements.tex [hunk 2]
```diff
@@ -4168,6 +4168,14 @@ \subsubsection{Outputs}\label{outputs-36-1}
...
 This output is the heat energy added after material layer N from the \hyperref[constructioninternalsource]{ConstructionProperty:InternalHeatSource} object. Only valid for the CondFD solution algorithm.
 
+\paragraph{CondFD EMS Heat Source Power After Layer}
+
+This output is the heat power added after material layer N from the EMS heat flux actuator (Component type: ``CondFD Surface Material Layer''; Control type: ``Heat Flux''). Only valid for the CondFD solution algorithm.
+
+\paragraph{CondFD EMS Heat Source Energy After Layer}
+
+This output is the heat energy added after material layer N from the EMS heat flux actuator (Component type: ``CondFD Surface Material Layer''; Control type: ``Heat Flux''). Energy is aggregated on the electricity meter and is only valid for the CondFD solution algorithm.
+
```
Comment (mitchute) [inline doc/input-output-reference/src/overview/group-surface-construction-elements.tex:4178]: New EMS-specific output vars are now referenced here.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 3]
```diff
@@ -1811,6 +1811,8 @@ namespace HeatBalFiniteDiffManager {
         // EMS Conductivity Override
         if (condActuator.isActuated) {
             kt = condActuator.actuatedValue;
+            ktA1 = kt;
+            ktA2 = kt;
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:1815]: Bugfix: updating conductivity here with EMS override value.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 4]
```diff
@@ -2017,6 +2019,22 @@ namespace HeatBalFiniteDiffManager {
...
+                    if (specHeatActuator2.isActuated) {
+                        Cp2 = specHeatActuator1.actuatedValue;
+                    }
+
+                    // Update EMS internal variables
+                    surfFD.condNodeReport(i) = kt1;
+                    surfFD.specHeatNodeReport(i) = Cp1;
+                    surfFD.condNodeReport(i + 1) = kt2;
+                    surfFD.specHeatNodeReport(i + 1) = Cp2;
+
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:2037]: Bugfix: updating k and cp here with EMS override values.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 5]
```diff
@@ -2061,26 +2079,16 @@ namespace HeatBalFiniteDiffManager {
                         kt1 = condActuator1.actuatedValue;
                     }
 
-                    // EMS Conductivity 2 Override
-                    if (condActuator2.isActuated) {
-                        kt2 = condActuator2.actuatedValue;
-                    }
-
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:2068]: Unused code, removing.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 6]
```diff
@@ -2061,26 +2079,16 @@ namespace HeatBalFiniteDiffManager {
...
                     // EMS Specific Heat 1 Override
                     if (specHeatActuator1.isActuated) {
                         Cp1 = specHeatActuator1.actuatedValue;
                     }
 
-                    // EMS Specific Heat 2 Override
-                    if (specHeatActuator2.isActuated) {
-                        Cp2 = specHeatActuator2.actuatedValue;
-                    }
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:2077]: Unused code, removing.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 7]
```diff
@@ -2061,26 +2079,16 @@ namespace HeatBalFiniteDiffManager {
...
-
                     // Update EMS internal variables
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).condNodeReport(i) = kt1;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).specHeatNodeReport(i) = Cp1;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).condNodeReport(i + 1) = kt2;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).specHeatNodeReport(i + 1) = Cp2;
+                    surfFD.condNodeReport(i) = kt1;
+                    surfFD.specHeatNodeReport(i) = Cp1;
+                    surfFD.condNodeReport(i + 1) = kt2;
+                    surfFD.specHeatNodeReport(i + 1) = Cp2;
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:2091]: Renaming with already-existing local reference.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc [hunk 8]
```diff
@@ -2178,10 +2186,10 @@ namespace HeatBalFiniteDiffManager {
...
 
                     // Update EMS internal variables
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).condNodeReport(i) = kt1;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).specHeatNodeReport(i) = Cp1;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).condNodeReport(i + 1) = kt2;
-                    state.dataHeatBalFiniteDiffMgr->SurfaceFD(Surf).specHeatNodeReport(i + 1) = Cp2;
+                    surfFD.condNodeReport(i) = kt1;
+                    surfFD.specHeatNodeReport(i) = Cp1;
+                    surfFD.condNodeReport(i + 1) = kt2;
+                    surfFD.specHeatNodeReport(i + 1) = Cp2;
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalFiniteDiffManager.cc:2192]: Renaming with already-existing local reference.

**Review by mitchute — COMMENTED** [review]
I boneheadedly missed that the heat flux with/without EMS actuated k and cp values were not matching before 😒 . This is due to missing updating k and cp in two places with the EMS overrides. The heat flux values now match.

The plot below shows the direct and ems heat flux values for one surface. "Direct" values come from setting k and cp directly in the IDF, and the "ems" values come from setting k and cp to different values in the IDF and then modifying them back to the original values using EMS.

Screen Shot 2022-01-02 at 1 16 32 PM

File: doc/ems-application-guide/src/ems-actuators/thermal-envelope.tex (hunk 1 above)
Comment (Myoldmopar) [reply doc/ems-application-guide/src/ems-actuators/thermal-envelope.tex:113]: :+1:

File: doc/input-output-reference/src/overview/group-surface-construction-elements.tex (hunk 2 above)
Comment (Myoldmopar) [reply doc/input-output-reference/src/overview/group-surface-construction-elements.tex:4178]: Good addition.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/HeatBalFiniteDiffManager.cc:1815]: Yep, good call.

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc (hunk 5 above)
Comment (Myoldmopar) [reply src/EnergyPlus/HeatBalFiniteDiffManager.cc:2068]: Awesome

File: src/EnergyPlus/HeatBalFiniteDiffManager.cc (hunk 7 above)
Comment (Myoldmopar) [reply src/EnergyPlus/HeatBalFiniteDiffManager.cc:2091]: Nice.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Great fix here, the change makes sense, and some extra cleanup along the way.  CI is happy with it except the license year, which is fixed separately.  Not wasting more cycles just for that.  Merging this in, thanks @mitchute !

**Pull request merged**